eodatasets3/scripts/tostac.py

```diff
@@ -5,72 +5,218 @@
 from datetime import datetime
 from functools import partial
 from pathlib import Path
-from typing import Iterable
+from typing import Iterable, Dict
 from uuid import UUID
+from urllib.parse import urljoin
 
 import click
 import pyproj
-from click import echo
+import requests
+from jsonschema import validate
 from shapely.geometry.base import BaseGeometry
 from shapely.ops import transform
 
 from eodatasets3 import serialise
+from eodatasets3.model import DatasetDoc
 from eodatasets3.ui import PathPath
 
 
+# Mapping between EO3 field names and STAC properties object field names
+MAPPING_EO3_TO_STAC = {
+    "dtr:end_datetime": "end_datetime",
+    "dtr:start_datetime": "start_datetime",
+    "eo:gsd": "gsd",
+    "eo:instrument": "instruments",
+    "eo:platform": "platform",
+    "eo:constellation": "constellation",
+    "eo:off_nadir": "view:off_nadir",
+    "eo:azimuth": "view:azimuth",
+    "eo:sun_azimuth": "view:sun_azimuth",
+    "eo:sun_elevation": "view:sun_elevation",
+}
+
+
+def convert_value_to_stac_type(key: str, value):
+    """
+    Convert return type as per STAC specification
+    """
+    # In STAC spec, "instruments" have [String] type
+    if key == "eo:instrument":
+        return [value]
+    else:
+        return value
+
+
 @click.command(help=__doc__)
+@click.option(
+    "--stac-template",
+    "-t",
+    type=str,
+    help="File path to the STAC document template with static content",
+)
+@click.option(
+    "--stac-base-url", "-u", default="", type=str, help="Base URL of the STAC file"
+)
+@click.option(
+    "--explorer-base-url",
+    "-e",
+    default="https://explorer.dea.ga.gov.au",
+    type=str,
+    help="Base URL of the ODC Explorer",
+)
+@click.option(
+    "--validate/--no-validate",
+    default=True,
+    help="Flag it for stac document validation. By default flagged",
```

Comment:
What's the logic for validating here?

Reply:
Providing an option to validate or not during stac conversion